docs/guides/configuring-sso-ec-deployment.md

```diff
@@ -0,0 +1,103 @@
+---
+subcategory: ""
+page_title: "Configuring an SSO provider for a deployment"
+description: |-
+    An example of how to spin up a deployment preconfigured with a SAML provider
+---
+
+# Configuring a SAML provider for an Elastic Cloud Deployment
+
+A common use case for the Elastic Cloud Terraform provider i, is to spin up an elastic deployment preconfigured with an SSO Identity provider (SAML2.0 or OIDC based)
+
+Since such a configuration often results in a cyclic dependency, where the Elasticsearch cluster yaml file requires the cluster full url name, but the url has not been generated yet, we're going to use Elastic Cloud's alias feature.
+
+*Please note, this guide doesn't go through what's needed to be configured on the IDP side, as this can change for different identity providers*
+
+First, we'll create an "random_uuid" resource, like so:
+
+```terraform
+resource "random_uuid" "uuid" {}
+```
+
+This will allow us to use a randomally generated uuid in our deployment name, so that we can later use it within the alias and revieve a unique deployment name & url alias, even before our deploymemnt is actually created. 
```

Comment:
@tobio this isn't trivial to achieve though, which is why I'm using the uuid resource.
The problem here is a cyclic dependency between determining the value of the alias (or being able to generate well-known unique aliases *before* the deployment gets created) and setting it in the elasticsearch.yml configuration.

Reply:
Around 10% of this guide discusses generating and formatting a UUID so you can have a unique alias. Using a uuid is one way of having a unique alias, but it isn't the only way. IMO it's not a good default to be suggesting. 

IMO we're complicating the guide by adding more detail than is required to achieve the goal. We could simply say something along the lines of "we're going to set an alias, you'll have to make sure it's unique" rather than have to explain the `random_uuid` resource and resulting string formatting. I'd expect, for most customers `<org-name>-<deployment-name>` would be likely to be unique without the extra trailing noise.

Reply:
@tobio your assumption is correct only if you ever generate a single deployment.. which isn't the case for users who automate things with TF, for the most part.

Again, I want to "spoon feed" our customers and field, since I'm sure not "solving for alias uniqueness" in this guide, will just leave (mostly myself) with more questions about how to configure SAML on an EC Deployment...

Reply:
> which isn't the case for users who automate things with TF

It seems reasonable to expect they would have a different deployment name though. Either way, I don't think this is a hill to die on :). It's a valid approach, just maybe more verbose than I'd choose 👍